packages/loopring_v3/contracts/amm/AmplifiedAmmController.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+pragma solidity ^0.7.0;
+
+import "./IAmmController.sol";
+import "../amm/LoopringAmmPool.sol";
+import "../lib/Claimable.sol";
+import "../lib/MathUint.sol";
+import "../lib/MathUint96.sol";
+import "../thirdparty/SafeCast.sol";
+
+
+/// @author Brecht Devos - <brecht@loopring.org>
+contract AmplifiedAmmController is IAmmController, Claimable
```

Comment:
Once a pool has any LP token minted, changing its getAmplificationFactor value won't have any impact, except that SET_VIRTUAL_BALANCE can be transacted. I think I'm having a hard time figuring out why.

Reply:
> If AF == 1 then the virtual balances are the actual balances, so this can be done without changing the behavior of the pool and this is what will need to be done to update our current pools to the new virtual balance system, which is the main reason this function is implemented like it is. If AF != 1 then the pool was already created in the new system and no updates are necessary to the virtual balances after the pool has been created.

The current implementation does not allow changing the AF while the pool is being used for the safety of LPs.